flowapi/flowapi/qa_endpoints.py

```diff
@@ -0,0 +1,205 @@
+        required: true
+        schema:
+          type: string
+          format: date
+        in: query
+      responses:
+        '200':
+          description: Dates available for each event type.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  qa_checks:
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        outcome:
+                          type: string
+                        type_of_query_or_check:
+                          type: string
+                        cdr_date:
+                          type: string
+                          format: date
+        '400':
+          description: Bad request
+        '401':
+          description: Unauthorized.
+        '404':
+          description: No QA checks of type specified found on date
+        '500':
+          description: Server error.
+      summary: Returns QA values for a given check type and call id between two dates
+    """
+    current_user.can_get_qa()
+    return await get_qa_checks(
+        cdr_type, check_id, request.args.get("start_date"), request.args.get("end_date")
+    )
+
+
+@blueprint.route("/<cdr_type>/<check_id>/<check_date>")
+@jwt_required
+async def get_qa_on_date(cdr_type, check_id, check_date):
+    """
+    Returns QA values for a given check type and call id on a date
+    ---
+    get:
+      parameters:
+      - name: cdr_type
+        in: path
+        required: true
+        schema:
+          type: string
+      - name: check_id
+        in: path
+        required: true
+        schema:
+          type: string
+      - name: check_date
+        in: path
+        required: true
+        schema:
+          type: string
+          format: date
+      responses:
+        '200':
+          description: Dates available for each event type.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  qa_checks:
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        outcome:
+                          type: string
+                        type_of_query_or_check:
+                          type: string
+                        cdr_date:
+                          type: string
+                          format: date
+        '400':
+          description: Bad request
+        '401':
+          description: Unauthorized.
+        '404':
+          description: No QA checks of type specified found on date
+        '500':
+          description: Server error.
+      summary: Returns QA values for a given cdr and check type on a given date
+    """
+    current_user.can_get_qa()
+    return await get_qa_checks(cdr_type, check_id, check_date, check_date)
```

Comment:
My feeling is this should really return just the value + nicer if the 404 message is something like: Check type {check_id} not found for cdr type {cdr_type} on date {check_date}

Reply:
Just confirming; 
```
CREATE VIEW etl.deduped_post_etl_queries AS (
    SELECT DISTINCT ON (cdr_date, cdr_type, type_of_query_or_check)
           cdr_date, cdr_type, type_of_query_or_check, outcome, optional_comment_or_description, timestamp
    FROM etl.post_etl_queries
    INNER JOIN etl.available_dates
    USING (cdr_date, cdr_type)
    ORDER BY cdr_date, cdr_type, type_of_query_or_check, timestamp DESC
);
```
If the same query has been run on the same cdr type on the same day but with a different timestamp, this would enforce that the latest output would be the last time that particuar check was run? Or in other words, `check_id`, `cdr_type` and `check_date` form a unique key inside this view?

Reply:
Yup.